### PR TITLE
CamScanner Script

### LINKED ISSUE #640: Python-camscanner
I want to contribute a script written in python to convert images into camscanner like images. I have made that script using opencv-python. It can be used for making notes

### PR DESCRIPTION
**Pull request by SmritiAgrawal04**
# Description

A script to generate scanned copy of the original image is made.

Fixes #640

## Type of change

_Please delete options that are not relevant._

- [ ] New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] My code follows the style guidelines(Clean Code) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have created a helpful and easy to understand `README.md`
- [ ] My changes generate no new warnings

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Hi @SmritiAgrawal04 

Great Work! I have a few questions: Is the sole purpose of this Script to convert an Image to grayscale? 

If yes, can we think of some customizations to this, that can make it near to an actual camscanner?

File: Python/Camscanner/camscanner.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	# convert to gray
+	im = im/255.0
+	# enhance the image to scan it
+	scannedImage = cv2.pow(im,2.0)
+	cv2.imshow('Original Image',im)
+	cv2.imshow('Scanned Image',scannedImage)
+	
+	# scannedImageFile= filename.split('.')[0]+'_scanned.jpeg'
+	# print (scannedImageFile)
+	# cv2.imwrite(scannedImageFile, scannedImage)
```
Comment (rahulraikwar00) [inline Python/Camscanner/camscanner.py:18]: it is saving complete black png file !

**Review by rahulraikwar00 — COMMENTED** [review]
Unable to exit after running script! It gets stuck on the running script !

**Comment by SmritiAgrawal04** [comment]
No it also works to enhance the image as is done in a CamScanner. Yet I'll
look into it again and see what best can be done.😁

**Comment by SmritiAgrawal04** [comment]
Hi @vybhav72954 

I've changed the script and added 8 ways to scan the image, please check if any of these suffices our requirements. The script runs the same way as before- check the README. Also, @rahulraikwar00 the script now does not get stuck.

File: Python/Camscanner/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Generate Scanned Image
```
Comment (vybhav72954) [inline Python/Camscanner/README.md:1]: Hey @SmritiAgrawal04 Can we please have a more detailed README 😅

**Review by vybhav72954 — CHANGES_REQUESTED** [review]
@HarshCasper After this one change.

Looks good to me.

File: Python/Camscanner/README.md (hunk 2 above)
Comment (SmritiAgrawal04) [reply Python/Camscanner/README.md:1]: Done! Can you please check if this looks good now?

File: Python/Camscanner/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+   `pip install numpy`
+   `pip install matplotlib`
+   
+2. Run the script.
+
+    `python camscanner.py \<filename\>`
+    
+    
+## References:
+1. https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_imgproc/py_thresholding/py_thresholding.html
```
Comment (rahulraikwar00) [inline Python/Camscanner/README.md:24]: @SmritiAgrawal04 throwing error when clicked on the attachment Hyperlink 
Error - Non-Image content-type returned
quick fix: just remove the exclamation mark from front

**Review by rahulraikwar00 — CHANGES_REQUESTED** [review]
and everything LGTM 👍

**Review by rahulraikwar00 — APPROVED** [review]
Looks fine..!

**Comment by SmritiAgrawal04** [comment]
can we merge this pull request then?

**Comment by SmritiAgrawal04** [comment]
> can we merge this pull request then?

Hey @vybhav72954 !
I request you to please merge this pull request.

**Review by vybhav72954 — APPROVED** [review]

File: Python/Camscanner/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+2. https://medium.com/@anupriyam/basic-image-thresholding-in-opencv-5af9020f2472
```
Comment (HarshCasper) [inline Python/Camscanner/README.md:26]: Add them as hyperlinks, not plain links

File: Python/Camscanner/README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
++ And this is what done in the script, multiple threshold values are set up in order to give image a scanned and enhanced look.
+
+
+## To run the script:
+1. Install the dependencies by running following command in terminal.
+
+   `pip install opencv-python`
+   `pip install numpy`
+   `pip install matplotlib`
```
Comment (HarshCasper) [inline Python/Camscanner/README.md:16]: Add a `requirements.txt` file and add all of these dependencies there

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**